### PR TITLE
Task] back end testing

### LINKED ISSUE #2: [Task] Back-end Testing
### Description

123

### Plan

123

### PR DESCRIPTION
**Pull request by alexzhal1**
## Summary

- Closes #2 

---

## Checklist

- [ ] I made this PR to the `main` branch **of my fork (NOT the course instructors' repo)**.
- [ ] I see `base: main` <- `compare: <branch-name>` above the PR title.
- [ ] I edited the line `- Closes #<issue-number>`.
- [ ] I wrote clear commit messages.
- [ ] I reviewed my own diff before requesting review.
- [ ] I understand the changes I’m submitting.

### CODE REVIEW

File: backend/app/models/interaction.py [hunk 1]
```diff
@@ -37,4 +37,4 @@ class InteractionModel(SQLModel):
     learner_id: int
     item_id: int
     kind: str
-    timestamp: datetime
```
Comment (Riderufa1984) [inline backend/app/models/interaction.py:40]: wow

File: backend/tests/unit/test_interactions.py [hunk 2]
```diff
@@ -24,3 +24,58 @@ def test_filter_returns_interaction_with_matching_ids() -> None:
     result = _filter_by_item_id(interactions, 1)
     assert len(result) == 1
     assert result[0].id == 1
+
+
+def test_filter_returns_all_interactions_with_matching_item_id() -> None:
+    """Edge case: Multiple interactions share the same item_id."""
+    interactions = [
+        _make_log(1, 1, 5),
```
Comment (Riderufa1984) [inline backend/tests/unit/test_interactions.py:32]: coool

**Review by Riderufa1984 — APPROVED** [review]
very good
